modules/gapi/src/backends/oak/goak_media_adapter.cpp

```diff
@@ -0,0 +1,101 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#include <opencv2/gapi/oak/oak_media_adapter.hpp>
+
+#ifdef WITH_OAK_BACKEND
+
+namespace cv {
+namespace gapi {
+namespace oak {
+
+class OAKMediaAdapter::Priv final {
+public:
+    Priv() = default;
+    void setParams(cv::Size sz, OAKFrameFormat fmt,
+                   const unsigned char* data_ptr, size_t data_sz);
+    size_t getDataSize();
+
+    MediaFrame::View access(MediaFrame::Access access);
+    cv::GFrameDesc meta() const;
+
+    ~Priv() = default;
+
+private:
+    cv::Size m_sz;
+    OAKFrameFormat m_fmt;
+    const unsigned char* m_data_ptr;
+    size_t m_data_sz;
+};
+
+void OAKMediaAdapter::Priv::setParams(cv::Size sz, OAKFrameFormat fmt,
+                                      const unsigned char* data_ptr, size_t data_sz){
+    m_sz = sz;
+    m_fmt = fmt;
+    m_data_ptr = data_ptr;
```

Comment:
What do you suggest? I can add `nullptr` checks in our backend, since we won't be exposing this memory to user

Reply:
> I can add nullptr checks in our backend

nullptr check doesn't work usually, because pointer could became dangling in any time especially in multithreading scenario.
In case of we controlling input or initial data creation and passsing it to MediaAdapter here - i suggest to use reference counting (if available). Dangling objects is hard in troubleshooting  and must be forbidden on early design step. Sorry - i do not know your scenario, just use common words :)